nilearn/signal.py

```diff
@@ -731,8 +731,10 @@ def clean(
         )
 
     # Interpolation / censoring
+    extrapolate = kwargs.get('extrapolate', False)
```

Comment:
This should be an explicit parameter exposed in `signal.clean` and add documentation on this. 
Since the current behaviour is `extrapolate=True`, we need to warn user about this change in behaviour.
If we want the default behaviour to be `extrapolate=False`, this will need to enter deprecation cycle (@ymzayek please make sure I am correct here)

Reply:
suggestion:
- add the parameter and document it
- do not change default behavior
- in the butterworth case throw a warning when the value of the parameter may lead to the unwanted outcome that @jhuguetn encountered

Reply:
Fine with me. I initially wanted to keep compatibility and impact as less as possible in the current way major/frontend functions are currently defined.